enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,180 @@
+
+As described in [the *Motivation* section](#motivation), enabling things like "we'd like to block this edge for clusters born in 4.1 with the `None` platform".
+
+#### OpenShift Dedicated
+
+[OpenShift Dedicated][dedicated] (OSD) restricts the recommended update sets for managed clusters more aggressively than the default service.
+For example, OSD blocked edges into 4.3.19 temporarily while [rhbz#1838007][rhbz-1838007] was investigated, while the default service did not.
+With this enhancement, OSD-specific edge blocking could be accomplished with entries like:
+
+```yaml
+to: 4.3.19
+from: .*
+clusters:
+  # show unmanaged clusters the edge, but block it from managed clusters
+  prometheus: |
+    max by (_id) (0*subscription_labels{managed="true"})
+    or
+    max by (_id) (subscription_labels{managed="false"})
+```
+
+The query returns `{_id="..."}=0` results for `managed="true"` clusters (blocking the edge) and `{_id="..."}=1` results for `managed="false"` clusters (allowing the edge).
+Clusters with both `managed="true"` and `managed="false"` records in the current time window will fall into the safer "block" bucket (the `or` semantics are defined [here][PromQL-or]).
+
+### Risks and Mitigations
+
+There is a risk that [published graph-data information](distribute-secondary-metadata-as-container-image.md) which assumes access to Red Hat's internal Telemetry will block edges when consumed by update services without access to the internal Telemetry.
+The risk is small initially, when we have no such edges.
+If targeted blocking becomes widespread, a simple but tedious mitigation would be asking users to maintain their own graph-data (e.g. removing all targeted edge-blocking references) and build their own graph-data image.
+
+A more powerful but complicated fix would be documenting a process for running a local Telemetry aggregator and configuring the local update service to run queries against that aggregator.
+This documentation would need to cover mechanisms for adding any expected metrics that Red Hat currently adds in internal Telemetry which come up in targeted edge queries, such as [`subscription_labels`](#openshift-dedicated).
+
+## Design Details
+
+### Test Plan
+
+[The graph-data repository][graph-data] should grow a presubmit test to enforce as much of the new schema as is practical.
+Validating any PromQL beyond "it's a string" is probably more trouble than its worth, because consuming update services should [alert on and safely handle](#update-service-support-for-the-enhanced-schema) such misconfiguration.
+
+Unit-testing the behavior in [openshift/cincinnati][cincinnati] with a mock Prometheus should be sufficient for [update service support](#update-service-support-for-the-enhanced-schema).
+
+### Graduation Criteria
+
+This will be released directly to GA.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
```

Comment:
Another drawback is that we can't inform particular clusters that they have been blocked

Reply:
I will mention this, but I will also mention that blocking reasons may not be exhaustive, so reviewing the reason(s) and deciding it doesn't actually apply to your cluster says nothing about update safety for your cluster vs. the many other issues which probably affect the edge and target release which are not mentioned, a large portion of which are probably bugs that *nobody* is aware of.  So I don't think this information is all that actionable.

Reply:
Oh yeah, that doesn't change current situation - just to mention that its a long standing problem, which needs to be solved separately

Reply:
I don't think it actually needs to be solved, because actionable data around this seems unsolvable ;).  But discussed in detail with d2f0889 -> 1f64222.